Add reads to test_simulated_stack.py

dcolligan opened the pull request:
Issue #462

jeromekelleher commented on tests/unit/test_simulated_stack.py:
```diff
@@ -155,3 +159,29 @@ def testCallSetsSearch(self):
+        # search reads
+        path = utils.applyVersion('/reads/search')
+        request = protocol.SearchReadsRequest()
+        readGroupId = 'aReadGroupSet:one'
```
This is pretty brittle --- why not get the readGroupId from the readGroupSets object you have above?

jeromekelleher commented on tests/unit/test_simulated_stack.py:
```diff
@@ -155,3 +159,29 @@ def testCallSetsSearch(self):
+        responseData = protocol.SearchReadsResponse.fromJsonString(
+            response.data)
+        alignments = responseData.alignments
+        self.assertEqual(len(alignments), 2)
```
How do you know this is 2?

jeromekelleher commented:
I guess I'd like to see more automation here @dcolligan, and less coupling with the current state of the read simulator. However, the read simulator should be updated a lot too, as it's quite weak at the moment. We can open a separate issue on this if you prefer, and add in a requirement that this test code be generalised with that issue.

dcolligan commented:
@jeromekelleher pushed a commit addressing your concerns

If you want a more comprehensive revamp of the simulated code, please file another issue specifying what you'd like to see.

jeromekelleher commented:
Thanks @dcolligan --- a read simulator is a bigger task. This is good for now.

+1.

dcolligan commented:
@jeromekelleher merge me!
